apps/backend/src/modules/users/user.repository.ts

```diff
@@ -29,34 +44,133 @@ class UserRepository implements Repository {
 		return UserEntity.initialize(user);
 	}
 
-	public async findAll(): Promise<UserEntity[]> {
-		const users = await this.userModel.query().execute();
+	public async findAll(): Promise<UserGetByIdItemResponseDto[]> {
+		const users = await this.userModel
+			.query()
+			.select([
+				"users.id",
+				"firstName",
+				"lastName",
+				"users.group_id as groupId",
+				"email",
+				"groups.id",
+				"groups.key as groupKey",
+				"groups.name as groupName",
+			])
+			.innerJoin("groups", "users.group_id", "groups.id")
+			.leftJoin(
+				"groups_to_permissions",
+				"groups.id",
+				"groups_to_permissions.group_id",
+			)
+			.leftJoin(
+				"permissions",
+				"groups_to_permissions.permission_id",
+				"permissions.id",
+			)
+			.castTo<UserGetByIdItemResponseDto[]>()
+			.execute();
 
-		return users.map((user) => UserEntity.initialize(user));
+		return users;
 	}
 
-	public async findByEmail(email: string): Promise<null | UserEntity> {
-		const user = await this.userModel.query().where("email", email).first();
+	public async findByEmail(
+		email: string,
+	): Promise<null | UserGetByIdItemResponseDto> {
+		const user = await this.userModel
+			.query()
+			.where("email", email)
+			.withGraphJoined("group.permissions")
+			.first();
+
+		if (!user || !user.group || !user.group.permissions) {
+			return null;
+		}
```

Comment:
The group will always be present and permissions should be set to empty array if not present
```suggestion
		if (!user) {
			return null;
		}
```